### PR TITLE
WIP radiasoft/container-beamsim-jupyter#51: Update to JupyterLab 3.x

### PR DESCRIPTION
**Pull request by e-carlin**
Moving to 3.x JupyterLab changed the way extensions are packaged. They
moved from using NPM for distribution to using PyPi.
https://jupyterlab.readthedocs.io/en/stable/extension/extension_migration.html#jupyterlab-2-x-to-3-x

This package was never distributed through NPM, always a local
install. So, this packaging change has little impact on us. But, there
are some more subtle changes to things like file structure, removing
dependencies (ex webpack), and changing how the build steps works (ex
setup.py).

One major change is the extension no longer works in "classic" Jupyter
Notebooks. Only JupyterLab. Classic Notebooks are being deprecated
in favor of JupyterLab.

### CODE REVIEW

**Comment by e-carlin** [comment]
@robnagler these 3 PR's all need to merged/built at the same time.

**Comment by mkeilman** [comment]
I'll start on this today

**Comment by mkeilman** [comment]
FTY I was unable to run ```radia_run container-build``` because I don't have docker installed - I don't have enough disk space to install it. I used ```pip install jupyterlab``` etc. to get everything in place for testing

File: README.md [hunk 1]
```diff
@@ -5,39 +5,30 @@ Widget for visualizing 3D Radia models in a jupyter notebook
...
 
 For a development installation (requires npm),
 
     $ git clone https://github.com/radiasoft/jupyter-rs-radia.git
     $ cd jupyter-rs-radia
     $ pip install -e .
-    $ jupyter nbextension install --py --symlink --sys-prefix jupyter_rs_radia
-    $ jupyter nbextension enable --py --sys-prefix jupyter_rs_radia
-    $ jupyter labextension install js
+    $ jlpm install:extension
```
Comment (mkeilman) [inline README.md:23]: ```jlpm install:extension``` did not work for me:

```
[py3;@v4 jupyter-rs-radia]$ jlpm install:extension
yarn run v1.21.1
$ jupyter labextension develop --overwrite .
setup.py:41: DeprecatedWarning: create_cmdclass is deprecated as of 0.8 and will be removed in 1.0. "
Use `wrap_installers` to handle prebuild steps in cmdclass.
Use `get_data_files` to handle data files.
Use `include_package_data=True` and `MANIFEST.in` for package data.

  data_files_spec=data_files_spec
setup.py:45: DeprecatedWarning: install_npm is deprecated as of 0.8 and will be removed in 1.0. Use `npm_builder` and `wrap_installers`
  install_npm(HERE, build_cmd="build:prod", npm=["jlpm"]),
Installing /home/vagrant/src/mkeilman/jupyter-rs-radia/jupyter_rs_radia/labextension -> jupyter-rs-radia
Building extension in /home/vagrant/src/mkeilman/jupyter-rs-radia/jupyter_rs_radia
An error occurred.
FileNotFoundError: [Errno 2] No such file or directory: '/home/vagrant/src/mkeilman/jupyter-rs-radia/jupyter_rs_radia/package.json'
See the log file for details:  /tmp/jupyterlab-debug-py7p4a4f.log
error Command failed with exit code 1.
```

I'm not sure why it's looking in the jupyter_rs_radia directory.

```jupyter labextension install .``` worked though.

File: README.md [hunk 2]
```diff
@@ -5,39 +5,30 @@ Widget for visualizing 3D Radia models in a jupyter notebook
...
 
-When actively developing your extension, build Jupyter Lab with the command:
+When actively developing your extension, use watch to hot reload on changes:
 
-    $ jupyter lab --watch
+    $ jlpm watch
```
Comment (mkeilman) [inline README.md:27]: jlpm watch also failed:

```
[py3;@v4 jupyter-rs-radia]$ jlpm watch
yarn run v1.21.1
$ watch:labextension
/bin/sh: watch:labextension: command not found
error Command failed with exit code 127.
```

I don't know if that is related to the previously failed command

File: package.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        "build:labextension": "jupyter labextension build .",
+        "build:labextension:dev": "jupyter labextension build --development True .",
+        "build:prod": "jlpm run build:labextension",
+        "clean:all": "jlpm run clean:labextension",
+        "clean:labextension": "rimraf jupyter-rs-radia/labextension",
+        "install:extension": "jupyter labextension develop --overwrite .",
+        "prepare": "jlpm run build:prod",
+        "watch": "watch:labextension",
+        "watch:labextension": "jupyter labextension watch ."
+    },
```
Comment (mkeilman) [inline package.json:67]: Seems related to the jlpm failures I was seeing but I'm not familiar enough to say

**Review by mkeilman — COMMENTED** [review]
Despite the problems I had with jlpm, I was able to run jupyterlab and successfully run the examples

**Pull request merged**